### ISSUE TITLE
launch --desktop fails due to a missing pyotherside depenency

### ISSUE THREAD

**Issue by doniks** [bhdouglass/clickable#54]
Hi, 

I installed clickable from ppa on 18.04. created a new app from template (python-cmake) and then try

```
clickable
clickable launch --desktop
```

but it fails with following output:

```
$ clickable launch --desktop
non-network local connections being added to access control list
file:///home/peter/docs/devel/rochester/build/tmp/qml/Main.qml:4 module "io.thp.pyotherside" is not installed

Command '['docker', 'run', '-v', '/home/peter/docs/devel/rochester:/home/peter/docs/devel/rochester', '-v', '/tmp/.X11-unix:/tmp/.X11-unix', '-v', '/tmp/.docker.xauth:/tmp/.docker.xauth', '-v', '/tmp/clickable/share:/tmp/.local/share', '-v', '/tmp/clickable/cacge:/tmp/.cache', '-v', '/tmp/clickable/config:/tmp/.config', '-e', 'XAUTHORITY=/tmp/.docker.xauth', '-e', 'DISPLAY=:0', '-e', 'QML2_IMPORT_PATH=/home/peter/docs/devel/rochester/build/tmp/lib/x86_64-linux-gnu:/usr/local/nvidia/lib:/usr/local/nvidia/lib64', '-e', 'LD_LIBRARY_PATH=/home/peter/docs/devel/rochester/build/tmp/lib/x86_64-linux-gnu:/usr/local/nvidia/lib:/usr/local/nvidia/lib64', '-e', 'PATH=/usr/local/nvidia/bin:/bin:/usr/bin:/home/peter/docs/devel/rochester/build/tmp/bin:/home/peter/docs/devel/rochester/build/tmp/lib/x86_64-linux-gnu/bin:/home/peter/docs/devel/rochester/build/tmp', '-e', 'HOME=/tmp', '-e', 'OXIDE_NO_SANDBOX=1', '-w', '/home/peter/docs/devel/rochester/build/tmp', '-u', '1000', '--rm', '-i', 'clickable/ubuntu-sdk:15.04-amd64', 'bash', '-c', 'qmlscene qml/Main.qml']' returned non-zero exit status 255.
```

**Comment by doniks** [comment]
I guess it's this:

sudo apt install qml-module-io-thp-pyotherside

maybe add to docs?

**Comment by JBBgameich** [comment]
Could you try adding pyotherside to the dependencies section of your clickable.json?

**Comment by JBBgameich** [comment]
I'd rather add the pyotherside package to the container images, if they are also included in the Ubuntu Touch images. `clickable --desktop` executes the app inside the docker container, and the docker magic was just forking the base image and installing the dependency inside the container.

**Comment by JBBgameich** [comment]
I just looked at your log, and it looks like pyotherside should be included in the app itself ... Maybe there's really something to fix in the template.

**Comment by bhdouglass** [comment]
Currently the template has the pyothersides lib included with it (hopefully to go away when 16.04 of Ubuntu Touch is released). Could be an issue with the amd64 lib that's included, I'll have to poke the template some.

**Comment by bhdouglass** [comment]
Oh I see the problem. So you're running `clickable` and then `clickable launch --desktop`. So what's happening is clickable builds an armhf package and then you try to run that in desktop mode (which doesn't work because it expects the package to be amd64). So what you need to do instead is to run `clickable --desktop`, which will compile & run the package in an amd64 docker container.
